grafana/grafana
# Explore: Use Datasource Onboarding page when visiting without any datasource set up

**What is this feature?**

Like https://github.com/grafana/grafana/pull/58795, adds an onboarding page to Explore when a user navigates to it without having any datasource set up.

Fixes #57652

**Special notes for your reviewer**:

i renamed `Wrapper.tsx` (and test file) to `ExplorePage` to better describe it (and to avoid to create a "wrapper" of a "wrapper").
changes in `public/app/features/explore/ExplorePage.test.tsx` are unrelated, but they solve better issue that popped up by renaming the file.

for @grafana/dashboards-squad:
i changed the Datasource Onboarding component to be slightly more generic. i'm not super sure about the naming of `CTAText` and `onCTAClick` props, an alternative may be `dismissText` and `onDismiss` which may better reflect what that button is.
Also something kinda meh is importing it directly from dashboard, do you have any better idea on how we can share that component?

**How to test this:**
enable `datasourceOnboarding` feature flag and make sure there are no datasource set up in Grafana.
